### PR TITLE
16.0 mig endpoint route handler

### PR DESCRIPTION
**Pull request by lmignon**
[ ] Fix tests: Unfortunately from 16.0 it's no more allowed to call `new` on an abstract model...

### CODE REVIEW

**Comment by simahawk** [comment]
FTR we discussed the evolution of this tool during OCA days and I have to draft something based on a combo of #13 + #3

File: endpoint_route_handler/models/endpoint_route_handler.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        for the same path.
+        """
+        # TODO: add tests registering a fake model.
+        # However, @simahawk tested manually and it works.
+        # TODO: shall we check for route existance in the registry instead?
+        all_models = self._get_endpoint_route_consumer_models()
+        routes = [x["route"] for x in self.read(["route"])]
+        clashing_models = []
+        for model in all_models:
+            if model != self._name and self.env[model].sudo().search_count(
```
Comment (rousseldenis) [inline endpoint_route_handler/models/endpoint_route_handler.py:79]: Could you do the search before the loop with a read_group?

File: endpoint_route_handler/models/endpoint_route_handler.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                [("route", "in", routes)]
+            ):
+                clashing_models.append(model)
+        if clashing_models:
+            raise exceptions.UserError(
```
Comment (rousseldenis) [inline endpoint_route_handler/models/endpoint_route_handler.py:84]: In constraints, you should ValidationError instead

File: endpoint_route_handler/models/endpoint_route_handler.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            route = prefix + route
+        return route
+
+    _blacklist_routes = ("/", "/web")  # TODO: what else?
+
+    @api.constrains("route")
+    def _check_route(self):
+        for rec in self:
+            if rec.route in self._blacklist_routes:
+                raise exceptions.UserError(
```
Comment (rousseldenis) [inline endpoint_route_handler/models/endpoint_route_handler.py:179]: Same as above

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by simahawk** [comment]
I've re-done the migration on  https://github.com/OCA/web-api/pull/17 and w/ a workaround that makes it work `.new` on v15 and v16

**Comment by lmignon** [comment]
superseded by #17

**Pull request closed without merging**